tests/core/operations_test.py

```diff
@@ -0,0 +1,173 @@
+        sc.array(dims=['a'], values=[True, False, False]),
+        coords={'a': sc.arange('a', 3), 'b': sc.arange('a', 4)},
+    )
+    x = sc.DataArray(
+        sc.array(dims=['a'], values=[1, 2, 3], unit='m'),
+        coords={'a': sc.arange('a', 3)},
+    )
+    y = sc.array(dims=['a'], values=[-13, -11, -12], unit='m')
+    result = sc.where(condition, x, y)
+    expected = sc.DataArray(
+        sc.array(dims=['a'], values=[1, -11, -12], unit='m'), coords=x.coords
+    )
+    assert sc.identical(result, expected)
+
+
+def test_where_with_da_v_da() -> None:
+    condition = sc.DataArray(
+        sc.array(dims=['a'], values=[True, False, False]),
+        coords={'a': sc.arange('a', 3), 'b': sc.arange('a', 4)},
+    )
+    x = sc.array(dims=['a'], values=[1, 2, 3], unit='m')
+    y = sc.DataArray(
+        sc.array(dims=['a'], values=[-13, -11, -12], unit='m'),
+        coords={'a': sc.arange('a', 3)},
+    )
+    result = sc.where(condition, x, y)
+    expected = sc.DataArray(
+        sc.array(dims=['a'], values=[1, -11, -12], unit='m'), coords=y.coords
+    )
+    assert sc.identical(result, expected)
+
+
+def test_where_with_da_da_da() -> None:
+    condition = sc.DataArray(
+        sc.array(dims=['a'], values=[True, False, False]),
+        coords={'a': sc.arange('a', 3), 'b': sc.arange('a', 4)},
+    )
+    x = sc.DataArray(
+        sc.array(dims=['a'], values=[1, 2, 3], unit='m'),
+        coords={'a': sc.arange('a', 3)},
+    )
+    y = sc.DataArray(
+        sc.array(dims=['a'], values=[-13, -11, -12], unit='m'),
+        coords={'a': sc.arange('a', 3)},
+    )
+    result = sc.where(condition, x, y)
+    expected = sc.DataArray(
+        sc.array(dims=['a'], values=[1, -11, -12], unit='m'), coords=x.coords
+    )
+    assert sc.identical(result, expected)
+
+
+def test_where_with_v_da_da() -> None:
+    condition = sc.array(dims=['a'], values=[True, False, False])
+    x = sc.DataArray(
+        sc.array(dims=['a'], values=[1, 2, 3], unit='m'),
+        coords={'a': sc.arange('a', 3)},
+    )
+    y = sc.DataArray(
+        sc.array(dims=['a'], values=[-13, -11, -12], unit='m'),
+        coords={'a': sc.arange('a', 3)},
+    )
+    result = sc.where(condition, x, y)
+    expected = sc.DataArray(
+        sc.array(dims=['a'], values=[1, -11, -12], unit='m'), coords=x.coords
+    )
+    assert sc.identical(result, expected)
+
+
+def test_where_condition_coords_must_be_superset_of_others() -> None:
+    condition = sc.DataArray(
+        sc.array(dims=['a'], values=[True, False, False]),
+        coords={'a': sc.arange('a', 3)},
+    )
+    x = sc.DataArray(
+        sc.array(dims=['a'], values=[1, 2, 3], unit='m'),
+        coords={'a': sc.arange('a', 3), 'b': sc.arange('a', 4)},
+    )
+    y = sc.DataArray(
+        sc.array(dims=['a'], values=[-13, -11, -12], unit='m'),
+        coords={'a': sc.arange('a', 3), 'b': sc.arange('a', 4)},
+    )
+
+    # condition is missing coord
+    with pytest.raises(sc.DatasetError):
```

Comment:
Suggest using
```suggestion
    with pytest.raises(sc.DatasetError, match='<something>'):
```
especially since we are trying to detect multiple different exceptions.